### DISCUSSION TITLE
AttributeError: 'Document' object has no attribute 'pageCount'

### DISCUSSION THREAD

**Discussion by 88arvin** [PaddlePaddle/PaddleOCR#8972] [General]
While executing the PDF code, I got the following error:

AttributeError: 'Document' object has no attribute 'pageCount'

**Comment by yonglee7015** [comment]
me too, it seems fitz already update this function to page_count . But I don't know how to fix it.

**Comment by LDOUBLEV** [comment]
try another version of PyMuPDF (<1.21.0)

**Reply by yonglee7015** [reply]
!pip install --ignore-installed pymupdf==1.19.0  solved this issue.

Why paddleOcr not updated?

**Reply by sunyuhan19981208** [reply]
mark, it is useful

**Reply by westeast** [reply]
thanks , it`s usefull

**Comment by zztt** [comment]
I tried pymupdf==1.19.0，and it required 1.20.0. I installed pymupdf==1.20.0 and 1.21.0.  AttributeError: 'Document' object has no attribute 'pageCount'. There is no way to deal with pdf files.

**Reply by yonglee7015** [reply]
Are you sure it required 1.20.0? version must be lower then 1.20.0. Cos pageCount only for version lower than 1.20.0

**Comment by savikko** [comment]
One solution which seems to work:

Edit directly ppocr/utils/utility.py

From line 93->
```python
            for pg in range(0, pdf.page_count):
                page = pdf[pg]
                mat = fitz.Matrix(2, 2)
                pm = page.get_pixmap(matrix=mat, alpha=False)

                # if width or height > 2000 pixels, don't enlarge the image
                if pm.width > 2000 or pm.height > 2000:
                    pm = page.get_pixmap(matrix=fitz.Matrix(1, 1), alpha=False)
```

So change camelCases to snake_case:

* pageCount -> page_count
* getPixmap -> get_pixmap

**Reply by bolongliu** [reply]
thanks , it`s usefull.

**Reply by NNNorman** [reply]
2.7.1版的pdf2word.py 也有这个问题，105行开始按上述修改，即：
pageCount -> page_count
getPixmap -> get_pixmap

2.7.1 version: the same problem occure in pdf2word.py
So change from line 105:
pageCount -> page_count
getPixmap -> get_pixmap

**Comment by vlavorini** [comment]
Please also update the documentation

https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.7/doc/doc_en/quickstart_en.md

**Comment by Jacquelin803** [comment]
pdfDoc = fitz.open(pdfPath)
    for pg in range(pdfDoc.**_page_count_**):
        page = pdfDoc[pg]
        # rotate = int(0)
        zoom_x = 4  # (1.33333333-->1056x816)   (2-->1584x1224)
        zoom_y = 4
        mat = fitz.Matrix(zoom_x, zoom_y)
        pix = page.**_get_pixmap_**(matrix=mat, alpha=False)
        if not os.path.exists(imagePath):
            os.makedirs(imagePath)
        pix.**_save_**(imagePath + '/' + 'images_%s.png' % pg)

**Comment by TanishqSharma2022** [comment]
> While executing the PDF code, I got the following error:
> 
> AttributeError: 'Document' object has no attribute 'pageCount'

I found out the bug is just present in the documentation. The libraries have updated it. In the code just replace pageCount to page_count and getPixmap to get_pixmap
